lmdeploy/serve/async_engine.py

```diff
@@ -508,11 +497,16 @@ async def generate(
             self.id2step[str(session_id)] = step
         if gen_config is None:
             gen_config = GenerationConfig()
-        if type(gen_config) is GenerationConfig:
-            gen_config = EngineGenerationConfig.From(gen_config,
-                                                     self.tokenizer)
-        if gen_config.stop_words is None:
-            gen_config.stop_words = self.stop_words
+        gen_config.convert_stop_bad_words_to_ids(self.tokenizer)
```

Comment:
why not do it when building `EngineGenerationConfig`?

Reply:
I see. It was removed. But we may do it when building `GenerationConfig`.

Reply:
Convert the stop_words to stop_words_ids needs a tokenizer.

Reply:
How can we ensure that `convert_stop_bad_words_to_ids` is not called twice when the user has converted it outside the function?

Reply:
No, that would be okay if it does not affect the final result.